### PR TITLE
[refactor](jni) unified jni framework for jdbc catalog

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

This commit overhauls the JDBC connector logic within our project, transitioning from the previous mechanism of fetching data through JNI calls for individual `ResultSet` items to a more efficient and unified approach using the `VectorTable` data structure.

### Key Changes:
- **Old Approach**: Data was initially read from the JDBC `ResultSet` into a `List<Object[]>`. Then, our C++ code within `jdbc_connector` had to read data from each column separately, making JNI calls to Java for necessary data format conversions and alignment for memory access.
- **New Approach with VectorTable**: All data conversion is now handled in Java. The C++ side only requires knowledge of the `VectorTable` memory address to directly fill the `Block` structure.

### Benefits:
- **Simplified Logic**: The introduction of `VectorTable` significantly declutters the data fetching and conversion logic on the Backend (BE) side for JDBC Catalog. C++ code is now cleaner and more maintainable, with a clear separation of responsibilities.
- **Enhanced Efficiency**: Data transformation is localized within the Java layer, reducing the overhead of context switching between Java and C++ and minimizing JNI overhead.
- **Improved Performance**: Directly accessing `VectorTable` allows for faster data processing, better memory management, and overall improved read performance from JDBC sources.

This refactor is part of our ongoing commitment to improve the architecture and performance of our system. By streamlining the data path from JDBC to our internal data structures, we expect to see more responsive data operations and lower latency in data handling tasks.

### Impact:
The changes are expected to be backward compatible with existing JDBC data sources. However, thorough testing is recommended to ensure that all edge cases are handled correctly.

We encourage the community to test this new implementation and provide feedback on any issues or performance improvements observed.

### Pending Tasks:
- **Optimization for Specific Data Types**: Currently, data types such as JSONB, bitmap, and HLL are handled through a process where they are first read into a String Column and then cast appropriately. This is an intermediate solution, and there is an ongoing effort to find a more optimized, memory-aligned method that would eliminate the need for casting within the C++ layer.

We aim to address this in future updates to further enhance the efficiency and performance of our system when dealing with these complex data types. By doing so, we anticipate reducing the overhead and potential bottlenecks associated with the current casting process, thus streamlining the entire data flow from JDBC to our internal representations.

Community involvement is crucial in this phase. We welcome contributions and suggestions on how to best approach this optimization for specialized data types. Your input will be invaluable in shaping the next iteration of our JDBC connector.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by AshinGau — DISMISSED** [review]

**Comment by AshinGau** [comment]
LGTM

**Comment by AshinGau** [comment]
Please finish the `Proposed changes` in detail.

**Review by chenlinzhong — DISMISSED** [review]
LGTM

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by AshinGau — DISMISSED** [review]

**Review by BePPPower — APPROVED** [review]

File: fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/JdbcExecutor.java [hunk 1]
```diff
@@ -483,1071 +362,402 @@ private void setValidationQuery(DruidDataSource ds, TOdbcTableType tableType) {
...
+                    return createConverter(
+                            input -> {
+                                List<Object> list = parseArray(String.valueOf(input));
+                                return convertArray(list, columnType.getChildTypes().get(0));
+                            },
+                            List.class);
+                } else {
+                    return createConverter(
+                            input -> {
+                                List<Object> list = convertClickHouseArray(input);
```
[the remark this answers is not in the document]
Comment (zy-kkk) [reply fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/JdbcExecutor.java:655]: done

File: fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/JdbcExecutor.java [hunk 2]
```diff
@@ -483,1071 +362,402 @@ private void setValidationQuery(DruidDataSource ds, TOdbcTableType tableType) {
...
+                                return convertArray(list, columnType.getChildTypes().get(0));
+                            },
+                            List.class);
+                }
+            default:
+                throw new IllegalArgumentException(
```
[the remark this answers is not in the document]
Comment (zy-kkk) [reply fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/JdbcExecutor.java:661]: yes , not support now

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by AshinGau — APPROVED** [review]

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by LemonLiTree — APPROVED** [review]
LGTM
